modules/swagger-codegen/src/main/resources/php/api.mustache

```diff
@@ -122,22 +124,38 @@ class {{classname}} {
         // for HTTP post (form)
         $httpBody = $formParams;
       }
-
-      // authentication setting, if any
-      $authSettings = array({{#authMethods}}'{{name}}'{{#hasMore}}, {{/hasMore}}{{/authMethods}});
-
+      {{#authMethods}}{{#isApiKey}}
```

Comment:
They both work, the reason I moved it into the responsibility of the *Api classes is so that the client is not coupled to any specific api. 

For us, we'll want to have our central application talking to many microservice APIs. Ideally they would all share a single `ApiClient`, so any improvements would affect all the API calls, not just the ones for the API which recently generated them. In order to do that, we had to move anything that was specific to a particular API (such as how it gets authenticated), out of the core classes.

To the point of duplicated code, I agree, though since this is generated and shouldn't be hand-modified, I think this is an acceptable amount of duplication to have spread across the APIs. I can change it if you disagree, but would really like this change to stick :)